src/types/policyData.go

```diff
@@ -341,14 +341,14 @@ type PolicyYaml struct {
 
 // MatchSpec spec to match for defining policy
 type MatchSpec struct {
-	Name              string          `json:"name" yaml:"name"`
-	Precondition      []string        `json:"precondition" yaml:"precondition"`
-	Description       Description     `json:"description" yaml:"description"`
-	Yaml              string          `json:"yaml" yaml:"yaml"`
-	Spec              KnoxSystemSpec  `json:"spec,omitempty" yaml:"spec,omitempty"`
-	Kind              string          `json:"kind,omitempty" yaml:"kind,omitempty" bson:"kind,omitempty"`
-	KyvernoPolicySpec *kyvernov1.Spec `json:"kyvernoPolicySpec,omitempty" yaml:"kyvernoPolicySpec,omitempty"`
-	KyvernoPolicyTags []string        `json:"kyvernoPolicyTags,omitempty" yaml:"kyvernoPolicyTags,omitempty"`
+	Name              string                     `json:"name" yaml:"name"`
+	Precondition      []string                   `json:"precondition" yaml:"precondition"`
+	Description       Description                `json:"description" yaml:"description"`
+	Yaml              string                     `json:"yaml" yaml:"yaml"`
+	Spec              KnoxSystemSpec             `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Kind              string                     `json:"kind,omitempty" yaml:"kind,omitempty" bson:"kind,omitempty"`
+	KyvernoPolicy     *kyvernov1.PolicyInterface `json:"kyvernoPolicy,omitempty" yaml:"kyvernoPolicy,omitempty"`
```

Comment:
We are storing the complete policy now, instead of just the spec. This removes the problem of adding all annotations with code, annotations can be pulled from policy templates directly.